prow/crier/controller.go

```diff
@@ -166,27 +193,12 @@ func (r *reconciler) reconcile(log *logrus.Entry, req reconcile.Request) (*recon
 		return requeue, nil
 	}
 
-	log.Info("Reported job, now will update pj")
+	log.Info("Reported job(s), now will update pj(s)")
 	for _, pjob := range pjs {
-		reportedState := pjob.Status.State
-		// We have to retry here, if we return we lose the information that we already reported this job.
-		if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-			// Get it first, this is very cheap
-			if err := r.pjclientset.Get(r.ctx, req.NamespacedName, &pj); err != nil {
```

Comment:
Bug 2. This `Get` call gets the original PJ identified by `req.NamespacedName` rather than getting the `pjob` PJ.